src/loadConfigurationFile.js

```diff
@@ -60,5 +60,12 @@ module.exports = function(configPath) {
             throw new Error('Could not load required module loading for ' + chalk.underline(configPath));
         }
     }
-    return require(configPath);
-}
\ No newline at end of file
+    var config = require(configPath);
+    if (typeof config === 'function')
+        return config(require('minimist')(process.argv, { '--': true }).env || {});
```

Comment:
1) Unnecessary code duplication. 
2) Wrong code. There are 4 different permutations for function/object and default/no default.
3) Split this function. It's too long already.

It should be refactored along these lines: 

```
var configModule = require(configPath);
var configDefault = configModule && configModule.__esModule ? configModule.default : configModule;

var config = typeof configDefault === 'function' ? callConfigFunction(configDefault) : configDefault;
return config;
```